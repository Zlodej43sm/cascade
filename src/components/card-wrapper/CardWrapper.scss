@use '../../styles/colors';
@use '../../styles/typography';
@use '../../styles/mixins';
@use '../../styles/sizes';

.container {
  @include mixins.themeBoxShadow(colors.$base-darkest);
  padding: sizes.$base-3 sizes.$base-3 sizes.$base-5;
  background-color: colors.$base-lightest;
  border-radius: 10px;
}

.header {
  display: flex;
  margin-bottom: sizes.$base-3;
}

.icon {
  font-size: 42px;
  margin-right: sizes.$base-3;
  flex: sizes.$base-4 0 0;
  text-align: center;
  position: relative;
  top: - sizes.$base;
}

.icon,
.subTitle {
  color: colors.$cascade-secondary;
}

.title {
  @include typography.title-md;
  color: colors.$cascade-primary;
}

.subTitle {
  @include typography.sub-title;
}

.content {
  overflow: auto;
}
