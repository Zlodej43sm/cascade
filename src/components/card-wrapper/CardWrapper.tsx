import React from 'react';

import styles from './CardWrapper.scss';

export type CardWrapperProps = {
  icon?: React.ReactNode
  title?: string
  subTitle?: string
  children?: React.ReactNode
  width?: number | string
  height?: number | string
};

function CardWrapper({ icon, title, subTitle, children, width, height }: CardWrapperProps) {
  return (
    <section className={styles.container} style={{ width, height }} data-testid="card-wrapper-container">
      <header className={styles.header}>
        {icon && (
          <div className={styles.icon} data-testid="card-wrapper-icon">
            <span>{icon}</span>
          </div>
        )}
        <div>
          {title && (
            <div className={styles.title} data-testid="card-wrapper-title">
              {title}
            </div>
          )}
          {subTitle && (
            <div className={styles.subTitle} data-testid="card-wrapper-subtitle">
              {subTitle}
            </div>
          )}
        </div>
      </header>
      <div className={styles.content}>
        {children}
      </div>
    </section>
  );
}

export default CardWrapper;
