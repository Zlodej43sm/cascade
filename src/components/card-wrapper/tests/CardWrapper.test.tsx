import React from 'react';
import { render } from '@testing-library/react';

import { cardArguments } from '../__mocks__/CardWrapper.mock';
import CardWrapper from '../CardWrapper';

describe('CardWrapper', () => {
  it('Should match snapshot', () => {
    const { asFragment } = render(<CardWrapper {...cardArguments} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
