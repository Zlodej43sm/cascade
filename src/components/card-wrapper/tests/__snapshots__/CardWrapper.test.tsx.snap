// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardWrapper Should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="container"
    data-testid="card-wrapper-container"
    style="width: 100%; height: 150px;"
  >
    <header
      class="header"
    >
      <div
        class="icon"
        data-testid="card-wrapper-icon"
      >
        <span>
          #
        </span>
      </div>
      <div>
        <div
          class="title"
          data-testid="card-wrapper-title"
        >
          FY21 Revenue
        </div>
        <div
          class="subTitle"
          data-testid="card-wrapper-subtitle"
        >
          As of today, in comparison with last month
        </div>
      </div>
    </header>
    <div
      class="content"
    >
      Child content
    </div>
  </section>
</DocumentFragment>
`;
